charts/tileserver-api/templates/service.yaml

```
@@ -0,0 +1,28 @@
+---
+
+apiVersion: v1
+kind: Service
+
+metadata:
+  name: {{ include "tileserver.name" . }}
+  {{- if .Values.tileserver.api.service.annotations }}
+  annotations:
+    {{- toYaml .Values.tileserver.api.service.annotations | nindent 4 }}
+  {{- end }}
+  labels:
+    {{- include "api.labels" . | nindent 4 }}
+    {{- if .Values.tileserver.api.service.labels }}
+    {{- toYaml .Values.tileserver.api.service.labels | nindent 4 }}
+    {{- end }}
+
+spec:
+  {{- with .Values.tileserver.api.service }}
+  type: {{ .type }}
+  ports:
+    - port: {{ .port }}
+      name: http
+      targetPort: http
+  {{- end}}
```

Стоит явно указать протокол и дать возможность определять `externalTrafficPolicy`.
```suggestion
  type: {{ .type }}
  ports:
    - name: http
      protocol: TCP
      port: {{ .port }}
      targetPort: http
  {{- end }}
  {{- if .Values.tileserver.api.service.externalTrafficPolicy }}
  externalTrafficPolicy: {{ .Values.tileserver.api.service.externalTrafficPolicy }}
  {{- end }}
```